inout_test.go

```diff
@@ -0,0 +1,140 @@
+package fx_test
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/dig"
+	"go.uber.org/fx"
+	"go.uber.org/fx/fxtest"
+)
+
+func TestIn(t *testing.T) {
+	t.Parallel()
+
+	type in struct {
+		fx.In
+	}
+	assert.True(t, dig.IsIn(reflect.TypeOf(in{})), "Expected dig.In to work with fx.In")
+}
+
+func TestOut(t *testing.T) {
+	t.Parallel()
+
+	type out struct {
+		fx.Out
+	}
+	assert.True(t, dig.IsOut(reflect.TypeOf(out{})), "expected dig.Out to work with fx.Out")
+}
+
+func TestOptionalTypes(t *testing.T) {
```

Comment:
For two reasons:

* So that we can see how this is supposed to work from Fx's POV
* So that we can be assured Fx's integration with Dig works as expected